internal/http/services/owncloud/ocdav/ocdav_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+package ocdav
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	providerv1beta1 "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/utils"
+)
+
+/*
+The encodePath method as it is implemented currently is terribly inefficient.
+As soon as there are a few special characters which need to be escaped the allocation count rises and the time spent too.
+Adding more special characters increases the allocations and the time spent can rise up to a few milliseconds.
+Granted this is not a lot on it's own but when a user has tens or hundreds of paths which need to be escaped and contain a few special characters
+then this method alone will cost a huge amount of time.
+*/
+func BenchmarkEncodePath(b *testing.B) {
```

Comment:
seems like a good candidate for [fuzzing](https://go.dev/blog/fuzz-beta)

Reply:
Let's tackle that in another PR